File: LicentaInAngular.Server/DataLayer/Models/Comanda.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using LicentaInAngular.Server.DataLayer.Models;

namespace LicentaInAngular.Server.Models
{
    public class Comanda
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]

        public int IdComanda { get; set; }

        [ForeignKey("User")]  // ForeignKey to User (Admin)
        public int IdUser { get; set; }  // Nullable ForeignKey to User (Admin)
        public User? User { get; set; }

        [ForeignKey("Adresa")]
        public int IdAdresa { get; set; }
        public Adresa? Adresa { get; set; }

        public int? IdCard_CC { get; set; }

        [ForeignKey("IdCard_CC")]
        public Carduri Card_CC { get; set; }

        public DateTime? ETA { get; set; }
        [Required]
        public double PretTotal { get; set; }

        [Required]
        public bool IsPlaced { get; set; }
    }
}

File: licentainangular.client/src/app/models/Comanda.ts
import { User } from './User';
import { Adresa } from './Adresa';
import { Card_Plati } from './Card_Plati';

// models/Comanda.ts
export interface Comanda {
  IdComanda: number;
  IdUser: number;
  User?: any;          // navigational property, opțional
  IdAdresa: number;
  Adresa?: any;        // navigational property, opțional
  IdCard?: number;     // backend-l îl numește "IdCard_CC" sau similar
  Card?: any;          // navigational property, opțional
  ETA?: Date | string; // Data estimată de livrare
  PretTotal: number;   // => corespunde double PretTotal din C#
  IsPlaced: boolean;   // => corespunde bool IsPlaced
}

